source/StartPublishingWorkflowFunction/app.py

```
@@ -38,7 +38,30 @@ def lambda_handler(event, context):
 
         product_id = manifest_dict_flat['product_id']
         dataset_id = manifest_dict_flat['dataset_id']
-        asset_list = manifest_dict_flat['asset_list']
+        intial_asset_list = manifest_dict_flat['asset_list']
+        asset_list=[]
+
+        for entry in intial_asset_list:
+            asset_bucket=entry['Bucket']
+            prefix=entry['Key']
+            if (prefix.endswith('/')):
+                paginator = s3.get_paginator('list_objects_v2')
+                response_iterator = paginator.paginate(Bucket=asset_bucket,Prefix=prefix,PaginationConfig={'PageSize': 1000})
+                for page in response_iterator:
+                    logging.info(page)
+                    if 'Contents' not in page:
+                        return {

```

We should raise an exception here and let the final except-block catch the exception:

```
    except Exception as error:
        logging.error(f'lambda_handler error: {error}')
        logging.error(f'lambda_handler trace: {traceback.format_exc()}')
        result = {
            'Error': f'{error=}'
        }
        return json.dumps(result)
```

---

fixed in commit

---

not seeing the commit, maybe we are missing a push?